server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -0,0 +1,170 @@
+            validateParent(mailbox, updateRequest, mailboxSession);
+
+            updateMailbox(mailbox, updateRequest, mailboxSession);
+            responseBuilder.updated(update.getKey());
+        } catch (MailboxNameException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description(e.getMessage())
+                    .build());
+        } catch (MailboxNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The mailbox '%s' was not found", mailboxId))
+                    .build());
+        } catch (MailboxParentNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The parent mailbox '%s' was not found.", e.getParentId()))
+                    .build());
+        } catch (MailboxHasChildException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description("Cannot update a parent mailbox.")
+                    .build());
+        } catch (MailboxException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("anErrorOccurred")
+                    .description("An error occurred when updating the mailbox")
+                    .build());
+        }
+    }
+
+    private Mailbox getMailbox(String mailboxId, MailboxSession mailboxSession) throws MailboxNotFoundException {
+        return mailboxUtils.mailboxFromMailboxId(mailboxId, mailboxSession)
+                .orElseThrow(() -> new MailboxNotFoundException(mailboxId));
+    }
+
+    private void validateMailboxName(MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) throws MailboxNameException {
+        char pathDelimiter = mailboxSession.getPathDelimiter();
+        if (updateRequest.getName()
+                .filter(name -> name.contains(String.valueOf(pathDelimiter)))
+                .isPresent()) {
+            throw new MailboxNameException(String.format("The mailbox '%s' contains an illegal character: '%c'", updateRequest.getName().get(), pathDelimiter));
+        }
+    }
+
+    private void validateParent(Mailbox mailbox, MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) throws MailboxException, MailboxHasChildException {
+
+        if (updateRequest.getParentId().isPresent()) {
+            String newParentId = updateRequest.getParentId().get();
+            mailboxUtils.mailboxPathFromMailboxId(newParentId, mailboxSession)
+                    .orElseThrow(() -> new MailboxParentNotFoundException(newParentId));
+            if (mustChangeParent(mailbox.getParentId(), newParentId) && mailboxUtils.hasChildren(mailbox.getId(), mailboxSession)) {
```

Comment:
not sure it can't be fixed quickly but a lot of "utils" method calls belongs to objects : `updateRequest.mustChangeParent(mailbox)` , `mailbox.hasChildren`, etc